src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_help.py

```diff
@@ -22,6 +22,25 @@
           text: az {1} -n name -g MyResourceGroup
 """
 
+helps['vm format-secret'] = """
+    type: command
+    long-summary: >
+        Transform secrets into a form consumed by VMs and VMSS create via --secrets.
+    examples:
+        - name: Create a self-signed certificate with a the default policy and add to a virtual machine
+          text: >
+            az keyvault certificate create --vault-name vaultname -n cert1 \\
+              -p "$(az keyvault certificate get-default-policy)"
+
+            secrets=$(az keyvault secret list-versions --vault-name vaultname \\
+              -n cert1 --query "[?attributes.enabled]")
+
+            vm_secrets=$(az vm format-secret -s "$secrets") \n
+
+            az vm create -g group-name -n vm-name --admin-username deploy  \\
+              --image debian --secrets "$vm_secrets"
```

Comment:
There is non-trivial work to be done to format a key for use in vm secrets with or without cert store. Yes, we could accept only secret ids and fetch the vault resource id within `az vm create`, but that would change the input format for `--secret` when we need to input certificate store with a secret in `az vm create`.

I opt'd for a single format for input for `--secret` which matches the API paired with a transformation command which takes as input a list of secrets and optionally a certificate store. I think balances the tradeoffs between ease of use and flexibility to change the output of `az vm format-secret` if need arises due to API changes.

Do you have a different design in mind?